Autolink to CVEs (#396)

Xetnus opened the pull request:
This is the most basic implementation of autolinking to vulnerabilities. If this approach is accepted, then it can be applied to everything else in #427.

Concerns:
1. Icons cannot accompany the link with this approach, unless we customize Sanitize a bit more. Currently, Sanitize will remove the class attribute of any element that attempts to display an icon (like \<i> or \<span>), which prevents the icon from displaying.
2. We'll have to call the markdown method everywhere a CVE, tag, filepath, or git hash can appear in the text. Currently, the markdown method is only called on certain pages, not including the welcome page that displays all of the articles.

Thoughts?

andymeneely commented:
Good start. A couple of things I need:

* Unit tests. This is the kind of thing that's perfect for unit testing.
* Make sure there are unit tests for the specific concerns you have. I don't think they'll be a problem. 
* CVEs don't have to have 4 digits in that last bit. I'm pretty sure they have a spec somewhere, but I think it's now 6? Find the spec

Yeah so the markdown is only called in a few places, but I'd like to have it be called all over the place as we move forward. It was an experiment that worked, so now I'm thinking we embrace it.

Xetnus commented:
According to [this spec](https://cve.mitre.org/about/faqs.html#cve_id_syntax_change), the last part of the CVE has to be at least 4 digits long. There's no max limit so I changed the regex accordingly. Thanks for letting me know.

I added some unit tests to make sure CVEs autolink correctly. I didn't attempt to include any icons yet, but if we still want to do that, then I can work on it as part of #427. And maybe I can go through and add the call to markdown on a few more pages in that issue as well.
